doc/en/user/source/community/cog/mosaic.rst

```diff
@@ -0,0 +1,209 @@
+
+.. admonition:: curl request
+
+   ::
+
+       curl -u admin:geoserver -XPOST -H "Content-type: text/plain" --write-out %{http_code} -d "https://modis-vi-nasa.s3-us-west-2.amazonaws.com/MOD13A1.006/2018.01.01.tif" "http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/remote.imagemosaic"
+
+*Response*
+
+::
+
+   202 Accepted
+
+
+**Initializing the store (Listing available coverages)**
+  
+Once a prototype has been provided we need to initialize the store by querying it for the available coverages.
+
+
+.. admonition:: curl request
+
+   ::
+
+       curl -v -u admin:geoserver -XGET http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/coverages.xml?list=all
+
+*Response*
+
+.. code-block:: xml
+
+    <List>
+      <coverageName>modisvi</coverageName>
+    </list>
+
+
+**Configuring the coverage**
+
+Once we get the list of available coverages, we need to configure a coverage by sending the config through REST.
+
+
+
+.. admonition:: curl request
+
+   ::
+
+       curl -v -u admin:geoserver -XPOST -H "Content-type: text/xml" -d @"coverage.xml" "http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/coverages"
+
+
+where coverage.xml has this content:
+
+.. code-block:: xml
+
+  <coverage>
+    <name>modisvi</name>
+    <nativeName>modisvi</nativeName>
+    <enabled>true</enabled>
+    <metadata>
+      <entry key="time">
+        <dimensionInfo>
+          <enabled>true</enabled>
+          <presentation>LIST</presentation>
+          <units>ISO8601</units>
+          <defaultValue>
+            <strategy>MAXIMUM</strategy>
+          </defaultValue>
+        </dimensionInfo>
+      </entry>
+    </metadata>
+  </coverage>
+
+
+**Adding more granules**
+
+Now that we have a coverageStore ready and a coverage layer configured we can start adding more granules.
+
+
+.. admonition:: curl request
+
+   ::
+
+       curl -u admin:geoserver -XPOST -H "Content-type: text/plain" --write-out %{http_code} -d "https://modis-vi-nasa.s3-us-west-2.amazonaws.com/MOD13A1.006/2018.01.17.tif" "http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/remote.imagemosaic"
+
+
+**Setting Style**
+
+That MODIS data has 2 bands representing (Normalized Difference Vegetation Index) (NDVI) and Enhanced Vegetation Index (EVI). 
+Let's add this ndvi.sld style to apply a proper colormap to the NDVI band:
+
+.. include:: src/modisvi/ndvi.sld
+   :literal:
+
+
+.. admonition:: curl request to create the style
+
+   ::
+
+       curl -v -u admin:geoserver -XPOST -H "Content-type: text/xml" -d "<style><name>ndvi</name><filename>ndvi.sld</filename></style>" http://localhost:8080/geoserver/rest/styles
+
+.. admonition:: curl request to upload the content of the style
+
+   ::
+
+	   curl -v -u admin:geoserver -XPUT -H "Content-type: application/vnd.ogc.sld+xml" -d @ndvi.sld http://localhost:8080/geoserver/rest/styles/ndvi
```

Comment:
Works, maybe a bit verbose? Should be possible to create the style and upload its body in a single request:
https://github.com/geoserver/geoserver/blob/e701004e988a8d0a9b7d993090c9a96dcf155e12/src/restconfig/src/main/java/org/geoserver/rest/catalog/StyleController.java#L208

Reply:
I will retry with that.

I was using instructions from 
https://docs.geoserver.org/latest/en/user/rest/styles.html
with the zipped version but I was unable to have it working.

Reply:
@aaime all the feedbacks have been addressed.